src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -568,8 +568,13 @@ static Linker nativeLinker() {
      * {@code T}, then the size of the returned segment is set to {@code T.byteSize()}.
      * <p>
      * The returned method handle will throw an {@link IllegalArgumentException} if the {@link MemorySegment}
-     * representing the target address of the foreign function is the {@link MemorySegment#NULL} address.
-     * The returned method handle will additionally throw {@link NullPointerException} if any argument passed to it is {@code null}.
+     * representing the target address of the foreign function is the {@link MemorySegment#NULL} address. If an argument
+     * is a {@link MemorySegment}, whose corresponding layout is an {@linkplain GroupLayout group layout}, that needs to
+     * be accessed by the linker implementation, one of the exceptions specified by the
```

Comment:
```suggestion
     * representing the target address of the foreign function is the {@link MemorySegment#NULL} address. If an argument
     * is a {@link MemorySegment},whose corresponding layout is an {@linkplain GroupLayout group layout}, the linker might attempt to access the contents of the segment. As such, one of the exceptions specified by the
```

Reply:
I've tried to simplify a little by making the sentence more direct. Also, removed "linker implementation" which we have tried to stay clear from in other places.